File: WorkcraftCore/src/main/java/org/workcraft/gui/propertyeditor/ReactiveComponent.java
package org.workcraft.gui.propertyeditor;

import java.awt.Component;

import org.workcraft.dependencymanager.advanced.core.Expression;
import org.workcraft.util.Nothing;

public interface ReactiveComponent {
	/**
	 * @return
	 * The component itself oblivious of any required state changes.
	 */
	public Component component();
	/**
	 * @return
	 * The expression, evaluation of which updates the component state.
	 */
	public Expression<? extends Nothing> updateExpression();
}

File: Gui/src/main/scala/org/workcraft/gui/propertyeditor/ReactiveComponent.scala
package org.workcraft.gui.propertyeditor

import java.awt.Component
import org.workcraft.scala.Expressions._
import javax.swing.JPanel
import java.awt.BorderLayout

class ReactiveComponent(component: Component, updateExpression: AutoRefreshHandle) extends JPanel(new BorderLayout) {
  add(component, BorderLayout.CENTER)
}

object ReactiveComponent {
  def apply(component: Component, updateExpression: AutoRefreshHandle) = new ReactiveComponent(component, updateExpression)
}